src/resources/general/compiler-development.md

```diff
@@ -0,0 +1,43 @@
+# 🛠️ Compiler Development
+
+This is a collection of resources on compiler construction, programming language design, and related topics.
+
+## 📚 Compiler Resources
```

Comment:
I know some people like using emojis all over the place for online content but we haven't been doing that on this wiki so far. I'm open to considering it but I don't think we should do it on just one page.